src/executionTimingsReporter.ts

```diff
@@ -0,0 +1,128 @@
+import { HighResTimer, calculateElapsedTime } from "./utils";
+import { EventEmitter } from "events";
+
+export type TimingsReport = {[stepName: string]: [number] | number};
+
+const stepStartedEventName = 'stepStarted';
+const milestoneReachedEventName = 'milestoneReached';
+
+interface StepStartedEventArguments {
+    stepName: string;
+}
+
+interface MilestoneReachedEventArguments {
+    milestoneName: string;
+}
+
+export interface ObservableEvents {
+    Events: EventEmitter;
```

Comment:
Should this type be `StepStartedEventsEmitter` ?

Reply:
I think it's better to promise only the minimum strictly necessary in the interface/contract, so then the implementer is free to solve this in whichever way seems best. If this property is an EventEmitter the code will work. We don't currently need it to be any particular implementation.